### PR TITLE
Implemented GUI High Score

### PR DESCRIPTION
**Pull request by gabe2762**
Add the high score feature to the GUI. Also added checking the puzzle state in the hints command that way when no puzzle is being run, the hints window doesn't display.

### CODE REVIEW

File: MVC/view/GUI.py [hunk 1]
```diff
@@ -148,7 +150,7 @@ def __init__(self, parent, ctrl, game_controller):
         self.fileMenu.add_separator()
         self.fileMenu.add_command(label="Export Score",command=self.screenShot)
         self.fileMenu.add_separator()
-        self.fileMenu.add_command(label="Exit",command=self.exitPuzzle) 
+        self.fileMenu.add_command(label="Give up",command = self.giveUp)
```
Comment (tmhughes1) [inline MVC/view/GUI.py:153]: please capitalize Up

File: MVC/view/GUI.py [hunk 2]
```diff
@@ -157,6 +159,8 @@ def __init__(self, parent, ctrl, game_controller):
         self.helpMenu.add_command(label = "How to play",command = self.playInstructions)
         self.helpMenu.add_separator()
         self.helpMenu.add_command(label = "Hints",command = self.displayAll)
+        self.helpMenu.add_separator()
+        self.helpMenu.add_command(label = "High scores",command= self.displayHighScores)
```
Comment (tmhughes1) [inline MVC/view/GUI.py:163]: please capitalize Score

File: MVC/view/GUI.py [hunk 3]
```diff
@@ -548,13 +552,87 @@ def totHint(self):
...
 
-        # Displays all hints
-        self.hintDisplay("Hints",hint1,hint2,hint3,1000,1000)
+            # Displays all hints
+            self.hintDisplay("Hints",hint1,hint2,hint3,1000,1000)
+    
+    '''
+    Function that creates the pop up windows for high scores.
+    '''
+    def highScoreWindow(self,title,message1,width,height):
```
Comment (tmhughes1) [inline MVC/view/GUI.py:570]: please use proper spacing after a ,

File: MVC/view/GUI.py [hunk 4]
```diff
@@ -548,13 +552,87 @@ def totHint(self):
...
+        if(self.controller.controllerGetPuzzleState() == 0):
+                messagebox.showinfo("Error!", "No game started!")
+                return
+        else:
+            confirm = messagebox.askyesno("Give up", "Are you sure you want to give up?")
+            if confirm:
+                points = self.controller.controllerGetPoints()
+                game_id = self.controller.controllerGetGameID()
+                while True:
+                    player_name = simpledialog.askstring("Username", "Please enter your 3-character username:")
```
Comment (tmhughes1) [inline MVC/view/GUI.py:596]: add a space after :

**Review by tmhughes1 — CHANGES_REQUESTED** [review]
minor syntax changes: see comments.

**Review by tmhughes1 — APPROVED** [review]
Thanks for making changes.

**Review by SteamsDev — APPROVED** [review]
Thanks Gabe!

**Pull request merged**